sunbeam-python/sunbeam/plugins/observability/etc/deploy-grafana-agent-k8s/main.tf

```
@@ -0,0 +1,87 @@
+# Terraform manifest for deployment of Grafana Agent
+#
+# Copyright (c) 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+terraform {
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = "= 0.8.0"
+    }
+  }
+}
+
+data "terraform_remote_state" "cos" {
+  backend = var.cos-state-backend
+  config  = var.cos-state-config
+}
+
+resource "juju_application" "grafana-agent-k8s" {
+  name  = "grafana-agent-k8s"
+  model = var.model
+
+  charm {
+    name     = "grafana-agent-k8s"
+    series     = "jammy" # we may need to use "base" in the future
```

`series` is still available (though deprecated) in the latest release.  So we can make this comment more precise with something like:

```suggestion
    series   = "jammy"
    # after upgrading to juju provider >= 0.10.0, replace series with:
    # base = "ubuntu@22.04"
```